### PR TITLE
Auditcare Cleanup

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

https://dimagi-dev.atlassian.net/browse/SAAS-13717
## Technical Summary

Cleans up the migration code that exists for Auditcare.
## Safety Assurance
Very contained change affects auditcare migration code only.
### QA Plan

NA

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/auditcare/models.py
Comment (dannyroberts) [inline corehq/apps/auditcare/models.py:240]: What if you kept this and the migration in a separate open PR until 6 weeks have passed from the time that the changelog was announced on the forum (which I think it hasn't yet been https://forum.dimagi.com/c/developers/maintainer-announcements/18). I suggest this because once this model is deleted, the steps described in that changelog entry won't work anymore—they'd have to manually undo the django migration or something like that.

Comment (millerdev) [reply corehq/apps/auditcare/models.py:240]: @dannyroberts are you saying the change log entry was never published to the forum (and that should be done before this is merged)? FWIW, I don't see that change log entry on the page you linked.

Comment (dannyroberts) [reply corehq/apps/auditcare/models.py:240]: Yeah that's what I was saying. I don't think this changelog was ever announced. (I noticed it half-way through writing that sentence so it came out a bit unclear!)

Comment (millerdev) [reply corehq/apps/auditcare/models.py:240]: I think I agree. Another option is to not merge this PR until at least 6 weeks after the change log entry has been announced on the forum, which is probably less convenient for us, but possibly better for everyone else.

Comment (AmitPhulera) [reply corehq/apps/auditcare/models.py:240]: The announcement has been made on the forum and I am okay with both approaches. I agree with Daniel that waiting for 6 weeks would be more convenient for the users.

Comment (dannyroberts) [reply corehq/apps/auditcare/models.py:240]: Based on your last comment, I think this Monday will be 6 weeks

Comment (AmitPhulera) [reply corehq/apps/auditcare/models.py:240]: Yep. Today is the day.  😄 
Can I get approval on the PR so I can merge it @dannyroberts @millerdev ?

**Review by millerdev — APPROVED** [review]

**Pull request merged**
